ambari-server/src/main/java/org/apache/ambari/server/controller/internal/BlueprintConfigurationProcessor.java

```diff
@@ -1950,7 +1950,7 @@ public String updateForClusterCreate(String propertyName,
               return origValue;
             }
 
-            if (topology.isComponentHadoopCompatible(component)) {
+            if (isComponentNameNode() && topology.hasHadoopCompatibleService()) { // do not fail, could be a ONEFS installation
               return origValue;
```

Comment:
This look like a stricter check than a previuos one. In the prevoius check the component could be any of the hadoop (compatible) components, in this case it has to be a NAMENODE. Also if a component is a NAMENODE it implies there is a hadoop compatible service IMO.

Reply:
NAMENODE is not marked as hadoop compatible, that's why it failed originally. When we install ONEFS the SingleHostTopologyUpdater still triggers and the component name is set to NAMENODE even though there is no NAMENODE in the blueprint (because both HDFS and ONEFS have the same config types and ambari doesn't fully support this). So this condition is for ignoring the error when there is a hadoop compatible FS in the blueprint (ONEFS) and the topology updater was invoked with componentName=NAMENODE.

Reply:
And what about the other case when the component is hadoop compatible but not NAMENODE? This passed the original filter but not the new one.

Reply:
I guess the original and the new condition should be ORed to cover both cases.

Since the original change (AMBARI-22781) worked fine at the time, I'm curious why it's a problem now.

Reply:
It's not clear why it worked before. According to the commit history HDFS has never been marked as HCFS. Based on the current code the original condition is not needed because we never call the SingleHostTopologyUpdater with componentName=ONEFS_CLIENT and even if we called it it would work since its matchingGroupCount would be valid.